File: demo/src/main/java/project/example/demo/Model/Appointment.java
package project.example.demo.Model;

import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(schema = "appointmentId_p")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idapp;
    private int idpatient;
    private int iddoctor;

    @Temporal(TemporalType.TIMESTAMP)
    private Date date;
    private String statut;

    // Default constructor
    public Appointment() {
    }

    public Appointment(int idapp, int idpatient, int iddoctor, Date date, String statut) {
    }

    public int getIdapp() {
        return idapp;
    }

    public void setIdapp(int idapp) {
        this.idapp = idapp;
    }

    public int getIdpatient() {
        return idpatient;
    }

    public void setIdpatient(int idpatient) {
        this.idpatient = idpatient;
    }

    public int getIddoctor() {
        return iddoctor;
    }

    public void setIddoctor(int iddoctor) {
        this.iddoctor = iddoctor;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    @Override
    public String toString() {
        return "Appointment{" +
                "idapp=" + idapp +
                ", idpatient=" + idpatient +
                ", iddoctor=" + iddoctor +
                ", date=" + date +
                ", statut='" + statut + '\'' +
                '}';
    }
}

File: front/src/app/models/Appointment.ts
import { DatePipe } from '@angular/common';

export class Appointment {
  idapp: number;
  idpatient: number;
  iddoctor: number;
  date: Date;
  formattedDate: string;
   statut:string;
  constructor(idapp?: number, idpatient?: number, iddoctor?: number, date?: Date,statut?:string) {
    this.idapp = idapp || -1;
    this.idpatient = idpatient || -1;
    this.iddoctor = iddoctor || -1;
    this.date = date || new Date();
    this.formattedDate = Appointment.formatDate(this.date);
    this.statut=statut||'';
  }

  private static formatDate(date: Date): string {
    const datePipe = new DatePipe('en-US');
    return datePipe.transform(date, 'short') || 'lr';
  }
}
